WooCommerce/Classes/ViewRelated/Products/Edit Product/Bundled Products/Bundled Product List/BundledProductsListViewModel.swift

```diff
@@ -67,6 +81,26 @@ private extension BundledProductsListViewModel {
 
         return controller.fetchedObjects
     }
+
+    /// Synchronizes the products matching the provided Product Bundle Items, to retrieve their product images
+    ///
+    func synchronizeBundledProductsIfNeeded(for bundleItems: [ProductBundleItem], siteID: Int64, stores: StoresManager) {
+        // We only need to sync if the bundled products are missing images
+        guard bundledProducts.filter({ $0.imageURL == nil }).isNotEmpty else { return }
+
+        let action = ProductAction.retrieveProducts(siteID: siteID, productIDs: bundleItems.map { $0.productID }) { [weak self] result in
```

Comment:
We could just sync the products that are missing images, but I decided not to try to be too clever about it and sync all products. That way, if we're syncing some products we'll go ahead and sync all of them, in case any of the stored image URLs have changed.

Reply:
Yep, the expectation is that this is a relatively small set of products, so we won't be syncing multiple pages. We can revisit this if it ends up that product bundles have lots of bundled products in them.

Since I've already merged this one, I'll add a comment about this to #9177 before I merge it.